README.md

```diff
@@ -676,12 +674,14 @@ You can install an import map for your application using a `<script>` element, e
 ```
 
 ```html
-<script type="importmap" src="import-map.json"></script>
+<script type="importmap" src="import-map.importmap"></script>
 ```
 
+External import maps must use the MIME type `application/importmap+json`. (Why not reuse `application/json`? Doing so could [enable CSP bypasses](https://github.com/WICG/import-maps/issues/105).)
```

Comment:
"External import maps" means any import map downloaded from a different domain? Does this mean that "internal import maps" (from the same domain) do not need this MIME type?

I think that requiring that MIME type might be a huge barrier for adoption of import maps 😢 . Running a simple static server, or putting the file on a CDN, would no longer be good enough to serve an import map. Unless all popular static servers and CDNs are PR'ed to include a default mime type for `.importmap` files, which is enough work to push out adoption significantly.

Reply:
It means import maps that are not inline in the document, but instead pulled from another URL.

I'm not significantly concerned about adoption of external import maps, as internal import maps will generally be the better pattern anyway.

Reply:
Tools that generate import maps will generate import maps into a file or stdout stream by default. Development workflows with import maps will naturally want to not have to edit the HTML template for simple operations like adding a new package, and instead use external import maps for ease of use. Production workflows can inline import maps where appropriate certainly, but disregarding entire development workflows has very real consequences for adoption.

Reply:
> I'm not significantly concerned about adoption of external import maps, as inline import maps will generally be the better pattern anyway.

I disagree with this. For an ES module, it's nice to update the live version of your module without having to update an HTML file. Additionally, it's powerful for devs/organizations to "vend" a hosted version of their import map for consumption by other devs/organizations, which is something best done with an external import map imo.

Reply:
At the organizations where I've helped them set up import maps, they host their js files on a CDN and want to be able to update the import map via CI process without having to redeploy the web server hosting the HTML file.

Reply:
Inlining any resource is generally considered a bad practice atm; I’m not sure why import maps would be any different.

Reply:
Inlining resources is definitely not considered a bad practice. Like critical, above-the-fold CSS, import maps are best inlined.

I agree it can be easier for developers to not inline, given their workflows. The same is true for above-the-fold CSS. Regardless, it's better to inline.

Reply:
I'm pretty sure your position is not widely agreed upon, perhaps mine isn't either - at the very least, anything that can be inlined should be able to be externalized, and vice versa.

Reply:
https://www.sitepoint.com/how-and-why-you-should-inline-your-critical-css/

Reply:
A lone article from 2.5 years ago doesn't prove or disprove either of our claims. This very website we're on right now doesn't have anything inlined, for example.